src/coord/src/catalog/storage.rs

```diff
@@ -226,6 +228,125 @@ const MIGRATIONS: &[&dyn Migration] = &[
 
         Ok(())
     },
+    // Allows us to dynamically assign system IDs to all objects but funcs. Also allows us to
+    // track built-in object name to ID mapping.
+    //
+    // Introduced in v0.26.0
+    &"ALTER TABLE gid_alloc RENAME TO user_gid_alloc;
+
+    CREATE TABLE system_gid_alloc (
+        next_gid integer NOT NULL
+    );
+
+    -- Higher than all statically assigned Type IDs
+    INSERT INTO system_gid_alloc VALUES (2000);
+
+    CREATE TABLE system_gid_mapping (
```

Comment:
Just realized that it would probably be better to have two columns and make them both the primary key instead of concatenating the names. Thoughts?

Reply:
I just did it, I think it'll be slightly cleaner.